app/assets/javascripts/models/data/service_options.js.coffee

```diff
@@ -68,7 +68,11 @@ ns.ServiceOptions = do (ko, edsc = @edsc, KnockoutModel = @edsc.models.KnockoutM
         clickedMethod = m
         break
 
+
       echoformContainer = if @index then document.getElementsByClassName('access-form')[@index] else document.getElementsByClassName('access-form')[0]
+      if (typeof echoformContainer != 'undefined')
```

Comment:
Thanks!  

This is repairing a bug where the form generation was really flaky.  There were instances where you could keep pushing 'FtpPushPull' and the form would be generated with every button push so you could have dozens of forms.

This fixes it.